cellular_automata/one_dimensional.py

```diff
@@ -0,0 +1,93 @@
+'''
+Returns an image of 16 generations of
+one-dimensional cellular automata 
+based on a given ruleset number
+https://mathworld.wolfram.com/ElementaryCellularAutomaton.html
+'''
+# Formats inputted ruleset
+def format_ruleset(RULE_NUM):
+    '''
+    >>> format_ruleset(11100)
+    [0,0,0,1,1,1,0,0]
+    >>> format_ruleset(0)
+    [0,0,0,0,0,0,0,0]
+    >>> format_ruleset(11111111)
+    [1,1,1,1,1,1,1,1]
+    '''
+    RULE = [int(a) for a in RULE_NUM]
+    while True:
+        if len(RULE) == 8:
+            break
+        else:
+            RULE.insert(0, 0)
+    #returns formatted ruleset
+    return(RULE)
+
+# Defines the first generation of cells
+
+CELLS = [[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,
+          0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,]]
+
+# Mainloop
+
+def new_generation(CELLS,RULE,time):
+    new_row = []
+
+    for i in range(31):
+
+        # Gets neighbors of a cell
+
+        # If leftmost cell is in consideration
+
+        if i == 0:
+            left_neighbor = 0
+            right_neighbor = CELLS[time][i + 1]
+        elif i == 30:
+
+        # If rightmost cell is in consideration
+
+            left_neighbor = CELLS[time][i - 1]
+            right_neighbor = 0
+        else:
+
+        # All other cells
+
+            left_neighbor = CELLS[time][i - 1]
+            right_neighbor = CELLS[time][i + 1]
+
+        # Defines new cell in and adds it to the new generation
+
+        SITUATION = 7 - int(str(left_neighbor) + str(CELLS[time][i])
+                            + str(right_neighbor), 2)
+        new_row.append(RULE[SITUATION])
+    #returns new generation
+    return new_row
+
+def generate_image(CELLS):
```

Comment:
```suggestion
def generate_image(CELLS: list) -> Image:
    """
    Convert the cells into a PIL.Image and return it to the caller.
    """
```
This function should only generate an Image and return it to the caller.  Have the caller do the `.show()`.  This would make it easier to embed `generate_image()` into a larger program which might print the image, put on a web page, display it in a Jupyter Notebook, etc.